numpyro/contrib/control_flow/scan.py

```diff
@@ -278,7 +279,7 @@ def scan_wrapper(
     length,
     reverse,
     rng_key=None,
-    substitute_stack=[],
+    substitute_stack=None,
```

Comment:
This is something I spotted while adding hints. We would want to avoid having mutables as defaults.